examples/features/example_good_descriptor.cpp

```diff
@@ -0,0 +1,152 @@
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *
+ */
+
+/** \brief GOOD: a Global Orthographic Object Descriptor for 3D object recognition and manipulation.
+  * GOOD descriptor has been designed to be robust, descriptive and efficient to compute and use. 
+  * It has two outstanding characteristics: 
+  * 
+  * (1) Providing a good trade-off among :
+  *	- descriptiveness,
+  *	- robustness,
+  *	- computation time,
+  *	- memory usage.
+  * 
+  * (2) Allowing concurrent object recognition and pose estimation for manipulation.
+  * 
+  * \note This is an implementation of the GOOD descriptor which has been presented in the following papers:
+  * 
+  *	[1] Kasaei, S. Hamidreza,  Ana Maria Tomé, Luís Seabra Lopes, Miguel Oliveira 
+  *	"GOOD: A global orthographic object descriptor for 3D object recognition and manipulation." 
+  *	Pattern Recognition Letters 83 (2016): 312-320.http://dx.doi.org/10.1016/j.patrec.2016.07.006
+  *
+  *	[2] Kasaei, S. Hamidreza, Luís Seabra Lopes, Ana Maria Tomé, Miguel Oliveira 
+  * 	"An orthographic descriptor for 3D object learning and recognition." 
+  *	2016 IEEE/RSJ International Conference on Intelligent Robots and Systems (IROS), Daejeon, 2016, 
+  *	pp. 4158-4163. doi: 10.1109/IROS.2016.7759612
+  * 
+  * Please adequately refer to this work any time this code is being used by citing above papers.
+  * If you do publish a paper where GOOD descriptor helped your research, we encourage you to cite the above papers in your publications.
+  * 
+  * \author Hamidreza Kasaei (Seyed.Hamidreza[at]ua[dot]pt)
+  */
+
+#include <pcl/features/good.h>
+#include <pcl/io/pcd_io.h>
+#include <pcl/io/ply_io.h>
+#include <boost/filesystem.hpp>
+#include <pcl/io/auto_io.h>
+
+typedef pcl::PointXYZRGBA PointT;  
+typedef pcl::PointCloud<PointT>::Ptr PointCloudIn;
+
+
+int main(int argc, char* argv[])
+{  
+
+  if (argc != 2) 
+  {
+    std::cout << "\n Syntax is: example_good_descriptor <path/file_name.pcd>" << std::endl;
+    return 0;
+  }
+  
+  std::string object_path =  argv[1];  
+  pcl::PointCloud<PointT>::Ptr object (new pcl::PointCloud<PointT>);
+  if (pcl::io::load(object_path, *object)==-1)  
+  {
+    std::cout << "\n file extension is not correct. Syntax is: example_good_descriptor <path/file_name.pcd>  or example_good_descriptor <path/file_name.ply>" << std::endl;
+    return -1;
+  }
+  
+  std::vector<PointCloudIn> vector_of_projected_views;
+  PointCloudIn transformed_object (new pcl::PointCloud<PointT>);
+
+  Eigen::Matrix4f transformation;
+  pcl::PointXYZ center_of_bounding_box;
+  pcl::PointXYZ bounding_box_dimensions;
+  std::string order_of_projected_planes;
+    
+  // Setup the GOOD descriptor
+  const int number_of_bins = 5;   
+  const unsigned int lengh_of_descriptor = 3 * number_of_bins * number_of_bins; 
+  pcl::PointCloud<pcl::Histogram<lengh_of_descriptor> > object_description;
```

Comment:
:+1:
Also there is a convention to name constants with capital letters, e.g. `NUMBER_OF_BINS`.